nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -737,9 +737,10 @@ def connect(self) -> None:
             self.connected = True
 
     def screenshot(self, filename: str) -> None:
-        word_pattern = re.compile(r"^\w+$")
-        if word_pattern.match(filename):
-            filename = os.path.join(self.out_dir, f"{filename}.png")
+        if "." not in filename:
+            filename += ".png"
+        if "/" not in filename:
+            filename = os.path.join(self.out_dir, filename)
```

Comment:
Should this check `if not filename.starts_with(self.out_dir)`?

Reply:
I assume there's a reason it's possible to use a path that isn't in `out_dir`? Like maybe you want to use a different output?  Otherwise there's no reason for the code having been this way to begin with.